react-components/src/components/NodeCacheProvider/PointCloudAnnotationCache.ts

```diff
@@ -106,15 +113,32 @@ export class PointCloudAnnotationCache {
     const filteredAnnotationMapping = filterUndefined(annotationMapping);
 
     const uniqueAnnotationMapping = uniqBy(filteredAnnotationMapping, 'assetId');
+    const assetIds = uniqueAnnotationMapping.map((mapping) => mapping.assetId);
+    const assets = await this.fetchAssetForAssetIds(assetIds, sdk);
 
+    const annotationIdToAssetMap = new Map<number, Asset>();
+    assets.forEach((asset) => {
+      uniqueAnnotationMapping.forEach((mapping) => {
+        if (mapping.assetId === asset.id) {
+          annotationIdToAssetMap.set(mapping.annotationId, asset);
+        }
+      });
+    });
+    return annotationIdToAssetMap;
+  }
```

Comment:
This can be very expensive if we have a lot of assets. Can we e.g. create a map from asset ID to asset first, then create this final map?